src/agentlab/analyze/agent_xray.py

```diff
@@ -581,6 +582,8 @@ def update_chat_messages():
     global info
     agent_info = info.exp_result.steps_info[info.step].agent_info
     chat_messages = agent_info.get("chat_messages", ["No Chat Messages"])
+    if isinstance(chat_messages, Discussion):
+        return chat_messages.to_markdown()
     messages = []
     for i, m in enumerate(chat_messages):
         if isinstance(m, BaseMessage):  # TODO remove once langchain is deprecated
```

Comment:
The Discussion class deprecates a lot code pieces but I figured it might be safer to keep for a while